### ISSUE TITLE
Amiga 100 KS1.3 WB34.34 "Not a DOS disk in unit 0"

### ISSUE THREAD

**Issue by craziazkowboi** [niklasekstrom/amiga-par-to-spi-adapter#8]
Hi Folks,

I am having an issue in both Workbench 1.3 and Workbench 3.2.2 where my A595 (SDBox from Amiga68k.com) won't mount, but gives me an error "Not a DOS disk in unit 0".  then on my workbench I get "SD0:BAD".  If I try to format the disk, it fails to format as soon as it starts (Error during format) and the icon name changes to SD0:BUSY but never changes until I reboot.  Under Workbench 3.2.2 the icon shows up as SD0: Unreadable, and when I try to format it, I get "Couldn't format cylinder 0".

I have tried reformatting using SD Card Formatter on my Mac as well.  I am using a 2GB card and FAT16.

I have tried multiple power adapters, and all provide at least 2 amps - should be a lot more than needed.  I have tried older spisd.device drivers as well as the v2.2 just released.

My PAL Amiga 1000 (no piggyback) has a Parceiro II card, using a 68000 CPU, 8MB FastRAM, boots from the Parceiro SD Card.  I have recapped my Amiga, but everything is working, I tried a PiStorm for a bit and it worked without a single crash, but the A595 would not work, same as above.  I am currently using Kickstart v34.5 and Workbench v1.3 v34.34 and have BindDrivers enabled in my Startup-Sequence if that helps.

I have since returned to the standard 68000 CPU, no PiStorm.  The A595 SDbox did work a few weeks ago, and then just stopped working?  What am I missing?

If I open the A595, is there something I can check?

Thanks!

**Comment by salocinx** [comment]
As to my own rather extensive testing, there are big differences in the quality of MicroSD cards. I had all kind of issues with cheap, small capacity cards. Which MicroSD brand are you using?

I never had issues when using quality brands like SanDisk (bought from local store, not aliexpress etc...). I suggest to use a 32 or 64GB MicroSD and format it using FAT32 with a maximum partition size of less than 32 GB, using the newly released driver v2.2. Perhaps better to start with an 8GB partition until you have it run stable.

**Comment by craziazkowboi** [comment]
Thanks.  It is a 2GB Sandisk card, which was purchased on eBay, but it does work on my Mac.  I also have a Verbatim 16GB card with a 3.9GB FAT32 partition which works on my Mac and my Amiga Parceiro.  Both cards report the same errors.  I am also using the latest v2.2 spisd.device driver.

**Comment by salocinx** [comment]
These SanDisk's from aliexpress (and probably the same they sell rather cheap on eBay) are either fake or as I  suspect the ones that failed QC (quality control) in the factory and were intended for trash but someone is going to sell them. I have a lot of different MicroSD cards here, almost all work fine on the PC and Mac, but fail on the SDBox. The SDBox is still quite picky regarding this. But I never had issues with quality SanDisk cards bought from local stores. But I agree, that we have to find a way to make the SDBox to accept a wider range of cards.

**Comment by craziazkowboi** [comment]
The Verbatim 16GB SD Card was a store purchase.  I can eject the card and the icon disappears off the workbench, so something is working.  Are there any diagnostics I can run?  Refresh my firmware perhaps?

**Comment by salocinx** [comment]
I am unfortunately not aware of any diagnostic tool, maybe someone else has an idea.

PS.: You detect bad MicroSD cards when running speed tests on the PC/Mac. The cheap ones, I talked about, reach only around 1-2 MB/s read/write speed, but they are sold as 20 MB/s or more; although they seem to work on the PC/Mac, they are trash (same with cheap USB sticks).

**Comment by craziazkowboi** [comment]
The Verbatim card is a good card, both get reasonable speeds on the Mac or a PC.  I'm pretty confident it is not the SDCard, but there could be a residual problem in my Amiga 1000 - not sure how to test that, or I've done something to my A595?  Thanks for all the help - it is appreciated!

**Comment by salocinx** [comment]
Now I remember, is this the same Amiga 1000 where you thought the 68010 CPU is not compatible with the SDBox driver? Maybe you still have some issue with the A1000 itself, could be a good idea to use DiagROM and a parallel port loopback dongle to test the parallel port. Another way would be to test the parallel port with a Plipbox for example.

**Comment by craziazkowboi** [comment]
Yes, guilty :). I have since re-capped this Amiga 1000, and it has been behaving much better.  Right now I am running with a 68000 CPU, not the 68010.  

I have booted to OS 3.2.2 and popped my plipbox back on the parallel port, the plipbox works - I can do a search using UHCTools on aminet without issue, 0 packets dropped, 0 buffer overruns, running at 491,520 bits/second, using the same power adapter as what I am trying to use with the A595. 

Is that enough to be confident that my parallel port is working?  Happy to try something else.

**Comment by salocinx** [comment]
Yes, I guess this enough to be confident that the parallel port is working properly.

How about the 3 LEDs on the A595/SDBox?

- Red Power LED: Should be bright, if you start the Amiga without PSU attached, it's much dimmer.
- Yellow LED: Should shine steadily as soon as a MicroSD card is inserted.
- Green LED: Should flicker when Amiga accesses the MicroSD card.

The reason why I believe the MicroSD card is the problem, is that I got the same errors as you do, with some MicroSD cards.

One more thing; do you have the [FAT95.lha](https://aminet.net/package/disk/misc/fat95) package installed properly?

**Comment by craziazkowboi** [comment]
Yep, the LEDs are coming on as you have described.  The Green LED flicks on and off once at boot time, just before the Not a DOS disk window opens.  I have another store-purchased 32GB SanDisk card, it also comes up with the same error, and I have a 32Gb Gigastone SDCard, also reports the same error.  That is 4 different cards with the same error, the 16, and both 32Gb cards work well in the Parceiro, and in a Mac, a PC, and a Raspberry Pi.

I have also double-checked my FAT95 install, but it must be working as I have a FAT32 partition on my Parceiro SD Card and it mounts and opens without a problem.

My MountList file already has entries for SD0: and SD1:, so I renamed the SD0 file in DEVS: to A595 and then edited the file, changing SD0: to A595: and finally editing my Mount_SDCard file to use A595 instead of SD0.  This did not help either, same error.

**Comment by niklasekstrom** [comment]
It's interesting that there are two open issues in this GitHub repo and both relate to A1000.

I have no idea what could be happening (I never had the fortune of having an A1000), but what if the CIA chips are somehow different than the CIA chips on later Amigas? Do you happen to have an A500 and can try to take its CIA chips and temporarily move them over to the A1000, and see if that makes any difference? You need to move over both CIA chips in that case I guess, as both are used by the design.

**Comment by salocinx** [comment]
As I understand it, the adapter once worked on his A1000. The question is what happened in between.

I can re-test on my Amiga 1000 (PAL) again as well, but I never had issues on that particular machine. But who knows, perhaps there are different CIA chips out there.

@craziazkowboi: I have uploaded the [MicroSD Boot Disk WB1.3](https://amiga68k.com/wp-content/uploads/2023/03/MicroSD_BootDisk_v22_wb13.zip) with newest driver v2.2 for you as ZIP file. Please write the containing ADF to a physical floppy disk using ADFBlitzer or some similar program and boot off this disk. Then double-click the Mount_SDcard icon and see what happens.

**Comment by idrougge** [comment]
The first run of the A1000 had an earlier CIA version than the one common in the A500 and latter-date A1000 machines. The question is if there are any other differences than manufacturing-related ones, though.

**Comment by craziazkowboi** [comment]
@salocinx Thanks for the ADF!  I have copied this to my GOTEK Drive and booted - I get a couple of read/write errors, but ignoring those when I execute Mount_SDCard, I again get the "Not a DOS disk in unit 0".  

Noting the spisd.device file on the ADF is 7504 bytes, and the spisd.device downloaded from GITHub is 7844 Bytes.

I also  happen to have replaced both my CIA chips with ones from an Amiga 500, but the CIA chips should be the same.  Earlier, when the A595 was working, these same CIA chips were already installed.  

Perhaps something has gone wrong with my A595 hardware?  If I can figure out how to open the case, I can get a probe onto it.

**Comment by craziazkowboi** [comment]
Any tips on opening the A595 case?  It is very solidly built - it will not pry open easily and I don’t want top break it.

Thanks!

**Comment by salocinx** [comment]
The case has four screws under the label on the back. The label is attached with double-sided adhesive tape. If you take a pair of tweezers and lift one corner, you can slowly pull the label away. Then you might be able to stick it back on afterwards.

**Comment by craziazkowboi** [comment]
Thanks!

**Comment by salocinx** [comment]
You're welcome. Could you shortly describe the story behind the CIA chips in your A1000? Did you replace them with some of an A500? Did the SDBox work before and afterwards it stopped working? Would be nice to have some information to determine if we need to pay attention to this.

**Comment by craziazkowboi** [comment]
When I purchased my Amiga 1000, it had a blown CIA-B chip.  I purchased 2 replacement CIA chips from eBay, and replaced both CIA chips.  The A595 was working with these same chips but only ever with a 68000 CPU, one day it just stopped working.  I booted up, and saw this same SD0:Bad error.  Nothing had changed at this stage.

I ignored the A595 not working for a while while I was playing round with a PiStorm, and now I have removed the PiStorm and gone back to the 68000 CPU, but cannot get the A595 to work again.

This Amiga 1000 is not the one that ‘blew up’ with the large file transfer back in 2022.  This Amiga is a PAL (no piggyback) which I have re-capped both the power supply and the mainboard and has been very stable.  As previously mentioned, the Plipbox works without a hitch on this one, using the same power-adapter that I am using for the A595.

I cannot see any obvious burn marks etc after opening and inspecting the A595, it is well built.  I have not had time to get an ESR meter or the o-scope onto it yet.
